tests/unit/downloader/test_entity.py

```diff
@@ -0,0 +1,262 @@
+import uuid
+
+import pytest
+from assertpy import assert_that
+
+from hca_ingest.downloader.schema_url import SchemaUrl
+from hca_ingest.downloader.entity import Entity
+
+
+@pytest.fixture
+def domain_type():
+    return 'protocol'
+
+
+@pytest.fixture
+def concrete_type():
+    return 'sequencing_protocol'
+
+@pytest.fixture
+def schema(domain_type, concrete_type):
+    return SchemaUrl(f'https://schema.humancellatlas.org/type/{domain_type}/sequencing/10.1.0/{concrete_type}')
+
+
+@pytest.fixture
+def metadata_uuid():
+    return str(uuid.uuid4())
+
+
+@pytest.fixture
+def metadata_id() -> str:
+    return str(uuid.uuid4()).replace('-', '')
+
+
+@pytest.fixture
+def populated_entity(schema, metadata_uuid, metadata_id):
+    return Entity({
+        'content': {
+            'describedBy': schema.url
+        },
+        'uuid': {'uuid': metadata_uuid},
+        "_links" : {
+            "self" : {
+                "href": f"https://api.ingest.archive.data.humancellatlas.org/projects/{metadata_id}"
+            }
+        }
+    })
+
+
+def test_populated_entity(populated_entity, metadata_uuid, metadata_id, schema):
+    assert_that(populated_entity.uuid).is_equal_to(metadata_uuid)
+    assert_that(populated_entity.id).is_equal_to(metadata_id)
+    assert_that(populated_entity.schema).is_equal_to(schema)
```

Comment:
You could use [chaining](https://github.com/assertpy/assertpy#chaining) and [dynamic assertions](https://github.com/assertpy/assertpy#dynamic-assertions-on-objects) to make the test more concise.
```suggestion
        assert_that(populated_entity).has_uuid(metadata_uuid).has_id(metadata_id).has_schema(schema)
```

Reply:
awesome, will add these asap.